src/hotspot/share/opto/memnode.cpp

```diff
@@ -427,14 +427,21 @@ Node *MemNode::Ideal_common(PhaseGVN *phase, bool can_reshape) {
 // Used by MemNode::find_previous_store to prove that the
 // control input of a memory operation predates (dominates)
 // an allocation it wants to look past.
-bool MemNode::all_controls_dominate(Node* dom, Node* sub) {
-  if (dom == nullptr || dom->is_top() || sub == nullptr || sub->is_top())
+bool MemNode::all_controls_dominate(Node* dom, Node* sub, bool *dead_code) {
+  bool dummy_flag, &dead_code_flag = dead_code != nullptr ? *dead_code : dummy_flag;
```

Comment:
This line is hard to read. And I'm not sure that `dummy_flag` is initialized. If not, it could most of the time happen to be `false`, but also randomly `true` sometimes. I would put the two variable definitions on separate lines.

Reply:
Also, this looks like a nasty hack with the pointer variable and the `dummy_flag`. Can we not solve this differently somehow?